freqtrade/strategy/strategy_helper.py

```diff
@@ -83,3 +86,46 @@ def stoploss_from_open(open_relative_stop: float, current_profit: float) -> floa
 
     # negative stoploss values indicate the requested stop price is higher than the current price
     return max(stoploss, 0.0)
+
+
+class HelperMixin(IStrategy):
```

Comment:
I'd rather like to see these as helpers (like `merge_informative_pair` above) - not as mixin.

having this as mixin does only complicate things and reduce usability as it's unclear to the user what will happen / what the result will be as it'll have some obscure dependencies and side-effects (it automatically fetches the dataframe, which is only one usecase.

`timeframe` and `dataframe` should be arguments for the methods (not assumed to be `self.timeframe` and `get_analyzed_dataframe()`).
Otherwise, you can't use it to get the candle for an informative pair / timeframe - which greatly limits reusability of this.

Reply:
thanks for the feedback. going for a mixin wasn't my initial plan, but I realised I would need `self` anyways and instead of passing it in, I decided to do a mixin. BUT now a lot of stuff has cleared up after implemening and some refactoring it and together with this input: 
> timeframe and dataframe should be arguments for the methods (not assumed to be self.timeframe and get_analyzed_dataframe()).
Otherwise, you can't use it to get the candle for an informative pair / timeframe - which greatly limits reusability of this.

it will make a lot more sense to go the helper function route.

TBC